packages/flutter/lib/src/widgets/app.dart

```diff
@@ -23,15 +25,51 @@ import 'widget_inspector.dart';
 
 export 'dart:ui' show Locale;
 
+/// The signature of [WidgetsApp.localeListResolutionCallback].
+///
+/// A `LocaleListResolutionCallback` is responsible for computing the locale of the app's
+/// [Localizations] object when the app starts and when user changes the list of
+/// locales for the device.
+///
+/// The `locales` list is the device's preferred locales when the app started, or the
+/// device's preferred locales the user selected after the app was started. This list
+/// is in order of preference The `supportedLocales` parameter is just the value of
+/// [WidgetsApp.supportedLocales].
+///
+/// When a `LocaleListResolutionCallback` is provided, Flutter will first attempt to
+/// resolve the locale with the provided `LocaleListResolutionCallback`, and if the
+/// callback or result is null, will fallback to trying the [LocaleResolutionCallback].
+/// If both [LocaleResolutionCallback] and 'LocaleListResolutionCallback' may be left
+/// null or fail to resolve (return null), the fallback algorithm will be used.
```

Comment:
this paragraph probably belongs in the docs for the actual property, not the typedef, since we can't make that promise about the typedef. The typedef is just the signature for the callback, it could be used in any number of places with different semantics.